cibuildwheel/__main__.py

```diff
@@ -66,6 +66,7 @@ def main():
         platform = args.platform
     else:
         ci = strtobool(os.environ.get('CI', 'false')) or 'BITRISE_BUILD_NUMBER' in os.environ or 'AZURE_HTTP_USER_AGENT' in os.environ
+        platform = None
```

Comment:
This was a bug introduced in a previous PR of mine. The error message a few lines below doesn't get triggered, instead we get a `platform is being used before assignment` kind of message.

Reply:
A test shall probably be added for this.

Reply:
here is my suggestion for base 
```
def test_platoform(utils):
    # check if cibuildwheel fail if runing without CIBW_PATFORM outside CI
    env = os.environ.copy()
    if "CIBW_PLATFORM" in env:
        del env["CIBW_PLATFORM"]
    if "CI" in env:
        del env["CI"]
    if "BITRISE_BUILD_NUMBER" in env:
        del env["BITRISE_BUILD_NUMBER"]
    if "AZURE_HTTP_USER_AGENT" in env:
        del env["AZURE_HTTP_USER_AGENT"]
    with pytest.raises(subprocess.CalledProcessError)
        utils.cibuildwheel_run()
```
checking output can be done in way showed [here](https://github.com/joerick/cibuildwheel/blob/99f831807aaf83d31a1249f5318f077ac455a543/test/09_platform_selector_problem/cibuildwheel_test.py)

Reply:
That's not the focus of this PR, though?

Reply:
Given the various comments about running tests or `cibuildwheel` itself unadvertently on a dev machine, I'd say it's crucial to test.
My comment "A test shall probably be added for this" is more of a self reminder to create another PR for this. Probably a unit test (which allows for more finer grained things without having to parse output) rather than an integration test.

Reply:
Hmmm, myes, OK